private/buf/cmd/buf/command/registry/registrylogin/registrylogin.go

```diff
@@ -165,6 +169,17 @@ func inner(
 		}
 		token = string(data)
 	} else {
+		// Check if we already have a valid token for this remote and username.
+		if machine, err := netrc.GetMachineForName(container, remote); machine != nil && err == nil {
+			currentToken := machine.Password()
+			if err := isValidToken(ctx, container, remote, username, currentToken); err == nil {
+				alreadyLoggedInMessage := "Valid credentials found in %s.\n"
```

Comment:
Assuming we want to keep this PR largely as-is, I would want this error message to tell me something like `Valid credentials already exist at %s, if you'd like to change them, please run `buf registry logout`.

Come to think of it though, by forcing users to call `logout`, what you're forcing them to do is delete *all of their BSR credentials, including for any Pro/Enterprise instances. So now, instead of being able to overwrite my token for a given instance with `buf registry login`, I get an error, and I have no way to individually overwrite a token for a single BSR instance as my only option is to delete all tokens.

Reply:
Oh, logout deleting all is probably not desired. Will close, can reopen when there's a better case for it.